packages/scripts/src/send-email-blast.js

```diff
@@ -0,0 +1,154 @@
+  array.map(row => {
+    fulltext += `${row.name},${row.email}\n`;
+  });
+  fs.writeFileSync(outputFilePath, fulltext);
+}
+
+// get emails from csv file
+async function getEmails() {
+  const file = readFileSync('./in/emails.csv').toString();
+  // Split the string into rows, discard the columns row
+  return file
+    .split('\n')
+    .slice(1)
+    .map((rowString, index) => {
+      // Split each row into it's variables
+      let row = rowString.split(',');
+      return {
+        name: row[0].trim(),
+        email: row[1].trim(),
+        confirmation_id: row[2].trim(),
+      };
+    });
+}
+
+// remove duplicate email entries
+function findDuplicates(emails) {
+  let unique = [];
+  let result = [];
+  emails.forEach(email => {
+    if (!unique.includes(email.email)) {
+      unique.push(email.email);
+      result.push(email);
+    } else {
+      console.log('Duplicate found', JSON.stringify(email));
+    }
+  });
+  return result;
+}
+
+// send individual emails using AWS SES
+async function sendEmail(user) {
+  const { name, email, confirmation_id } = user;
+  // unsure of actual template body, using hardcoded html string for now
+  const mailOptions = {
+    from: process.env.MAIL_FROM,
+    to: [email],
+    subject: subjectOne,
+    body: templateOne,
+  };
+
+  if (!ses) return;
+
+  const params = {
+    Destination: {
+      ToAddresses: [...mailOptions.to],
+    },
+    Message: {
+      Body: {
+        Html: {
+          Charset: 'UTF-8',
+          Data: mailOptions.body,
+        },
+      },
+      Subject: {
+        Charset: 'UTF-8',
+        Data: mailOptions.subject,
+      },
+    },
+    //TO-DO: testing purposes, need to remove and replace with prod
+    Source: 'EHPRDoNotReply@dev.ehpr.freshworks.club',
+  };
+
+  try {
+    console.log(`Trying to Send Email to ${email}`);
+    const mail = await ses.sendEmail(params).promise();
+    sent++;
+    return mail;
+  } catch (e) {
+    // set max retry count for failed emails
+    if (MAX_RETRY > retryCount) {
+      console.log(`Will try sending to ${user.email} ${MAX_RETRY - retryCount} more time(s)`);
+      retryCount++;
+      await sendEmail(user);
+    } else {
+      errors.push(user);
+      console.error(e.message);
+    }
+  }
+}
+
+// function to setup and send emails
+async function sendEmails(max) {
+  let emails = await getEmails();
+  console.log(`Sending ${emails.length} emails has begun...`);
+
+  // guard to limit emails and remove ones over max limit
+  // outputs to csv file
+  if (emails.length > max) {
+    generateEmailCSV(emails.slice(max), `out/unsent_emails${Date.now()}.csv`);
+    emails = emails.slice(0, max);
+  }
+
+  emails = findDuplicates(emails);
+
+  if (emails.length > 0) {
+    // Set up promise pool to allow 10 emails to be processed concurrently at one time
+    await PromisePool.for(emails)
+      .withConcurrency(10)
+      .process(async e => {
+        await sendEmail(e);
+        retryCount = 0;
```

Comment:
`retryCount` is global, but each successful call resets it. As a result, we could keep trying to send an invalid email endlessly.

Reply:
the retryCount will error out once the MAX_RETRY hits 5 and move on to the next one, resetting the retry count back to 0, I've had success with it, in this case it's a `The security token included in the request is invalid` error

`Will try sending to alt.rl-cobf22ha@yopmail.com 5 more time(s)

Trying to Send Email 
Will try sending to alt.rl-cobf22ha@yopmail.com 4 more time(s)

Trying to Send Email 
Will try sending to alt.di-fa0zfde@yopmail.com 3 more time(s)

Trying to Send Email 
Will try sending to alt.di-fa0zfde@yopmail.com 2 more time(s)

Trying to Send Email 
Will try sending to alt.rl-cobf22ha@yopmail.com 1 more time(s)

Trying to Send Email 
The security token included in the request is invalid

Will try sending to alt.di-fa0zfde@yopmail.com 5 more time(s)

Trying to Send Email 
Will try sending to alt.di-fa0zfde@yopmail.com 4 more time(s)

Trying to Send Email 
Will try sending to alt.di-fa0zfde@yopmail.com 3 more time(s)

Trying to Send Email 
Will try sending to alt.rl-cobf22ha@yopmail.com 2 more time(s)

Trying to Send Email 
Will try sending to alt.di-fa0zfde@yopmail.com 1 more time(s)

Trying to Send Email 
The security token included in the request is invalid
`